lit/SwiftExpr/RecursiveClass.test

```diff
@@ -0,0 +1,43 @@
+// Test that recursive class instances work in the REPL.
+
+// RUN: %lldb --repl < %s | FileCheck %s
+
+class Foo {
+  var aFoo: Foo!
+  var x: String = "Hello World"
+
+  init() {
+  }
+}
+
+var a = Foo()
+// CHECK: a: Foo = {
+// CHECK-NEXT:   aFoo = nil
+// CHECK-NEXT:   x = "Hello World"
+// CHECK-NEXT: }
+
+a.aFoo = a
+a
+// CHECK: $R0: Foo = {
+// CHECK-NEXT:   aFoo = 0x{{[0-9a-fA-F]+}} {...}
+// CHECK-NEXT:   x = "Hello World"
+// CHECK-NEXT: }
+
+// Also test the more basic non-recursive case.
```

Comment:
The Recursive class test was testing a particular tricky bit of the REPL - that you could use an optional of a class in it's definition.  Combining it with the basic "can we define classes" tests seems confusing to me.  Now if I want to add another test of class behavior where would I put it?  We should either keep this one narrowly focused and make a generic class behavior test (by just changing the name of the test...) or make another test for collecting more general class behavior.

Reply:
Sure, I'll split it back out.